sdk/containerregistry/container-registry/src/containerRepository.ts

```diff
@@ -0,0 +1,346 @@
+  ): Promise<RepositoryProperties> {
+    const { span, updatedOptions } = createSpan("ContainerRepository-setProperties", {
+      ...options,
+      value: {
+        canDelete: options.canDelete,
+        canWrite: options.canWrite,
+        canList: options.canList,
+        canRead: options.canRead
+      }
+    });
+
+    try {
+      const properties = await this.client.containerRegistry.setProperties(
+        this.name,
+        updatedOptions
+      );
+      return {
+        ...properties,
+        writeableProperties: {
+          canDelete: properties.writeableProperties.canDelete,
+          canList: properties.writeableProperties.canList,
+          canRead: properties.writeableProperties.canRead,
+          canWrite: properties.writeableProperties.canWrite
+        }
+      };
+    } catch (e) {
+      span.setStatus({ code: SpanStatusCode.ERROR, message: e.message });
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Iterates manifests.
+   *
+   * Example usage:
+   * ```ts
+   * const client = new ContainerRegistryClient(url, credentials);
+   * const repository = client.getRepository(repositoryName)
+   * for await (const manifest of client.listManifests()) {
+   *   console.log("manifest: ", manifest);
+   * }
+   * ```
+   * @param options -
+   */
+  public listManifests(
+    options: ListManifestsOptions = {}
+  ): PagedAsyncIterableIterator<ArtifactManifestProperties> {
+    const iter = this.listManifestsItems(options);
+
+    return {
+      next() {
+        return iter.next();
+      },
+      [Symbol.asyncIterator]() {
+        return this;
+      },
+      byPage: (settings: PageSettings = {}) => this.listManifestsPage(settings, options)
+    };
+  }
+
+  private async *listManifestsItems(
+    options: ListManifestsOptions = {}
+  ): AsyncIterableIterator<ArtifactManifestProperties> {
+    for await (const page of this.listManifestsPage({}, options)) {
+      yield* page;
+    }
+  }
+
+  private async *listManifestsPage(
+    continuationState: PageSettings,
+    options: ListManifestsOptions = {}
+  ): AsyncIterableIterator<ArtifactManifestProperties[]> {
+    const orderby =
+      options.orderBy === "timeAsc"
+        ? "timeasc"
+        : options.orderBy === "timeDesc"
+        ? "timedesc"
+        : undefined;
+    if (!continuationState.continuationToken) {
+      const optionsComplete = {
+        ...options,
+        n: continuationState.maxPageSize,
+        orderby
+      };
+      const currentPage = await this.client.containerRegistry.getManifests(
+        this.name,
+        optionsComplete
+      );
+      if (currentPage.link) {
+        continuationState.continuationToken = currentPage.link.substr(
+          1,
+          currentPage.link.indexOf(">") - 1
+        );
+      } else {
+        continuationState.continuationToken = undefined;
+      }
+      if (currentPage.manifests) {
+        yield currentPage.manifests.map((t) => {
+          return {
```

Comment:
In Key Vault we like to put translations like this (type A to type B) in a "transformations" file. What we find important is not the name of the file, but the idea of separating the renames from the logic around the renames. Sometimes these renames happen more than once, so these functions can be re-used. Here's one for KV Keys: https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/keyvault/keyvault-keys/src/transformations.ts